src/encord_active/lib/encord/actions.py

```diff
@@ -70,83 +134,87 @@ def create_dataset(
-                            image_paths.append(img_path.as_posix())
-                            image_names.append(img_path.name)
-
-                        # Unfortunately the following function does not return metadata related to the uploaded items
-                        dataset.create_image_group(file_paths=image_paths, title=label_row["data_title"])
-
-                        # Since create_image_group does not return info related to the uploaded images, we should find its
-                        # data_hash in a hacky way
-                        _update_mapping(
-                            self.user_client, dataset_hash, label_row_hash, data_unit_hash, new_du_to_original
-                        )
-
-                elif label_row["data_type"] == DataType.VIDEO.value:
-                    video_path = list(label_row_structure.images_dir.glob(f"{data_unit_hash}.*"))[0].as_posix()
-
-                    # Unfortunately the following function does not return metadata related to the uploaded items
-                    dataset.upload_video(
-                        file_path=video_path, title=label_row["data_units"][data_unit_hash]["data_title"]
+                    new_du_to_original[new_data_unit_hash] = LabelRowDataUnit(label_row_hash, data_unit_hash)
+                    lrdu_mapping[LabelRowDataUnit(label_row_hash, data_unit_hash)] = LabelRowDataUnit(
+                        "", new_data_unit_hash
                     )
-
-                    # Since upload_video does not return info related to the uploaded video, we should find its data_hash
-                    # in a hacky way
-                    _update_mapping(self.user_client, dataset_hash, label_row_hash, data_unit_hash, new_du_to_original)
-
-                else:
-                    raise Exception(
-                        f'Undefined data type {label_row["data_type"]} for label_row={label_row["label_hash"]}'
-                    )
-
-                uploaded_label_rows.add(label_row_hash)
-
-            if progress_callback:
-                progress_callback((counter + 1) / filtered_dataset.shape[0])
-
-        return DatasetCreationResult(dataset_hash, new_du_to_original)
+                    uploaded_data_units.add(data_unit_hash)
+
+                if progress_callback:
+                    progress_callback(len(uploaded_data_units) / filtered_dataset.shape[0])
+        return DatasetCreationResult(dataset_hash, new_du_to_original, lrdu_mapping)
+
+    def create_ontology(self, title: str, description: str):
+        ontology_dict = json.loads(self.project_file_structure.ontology.read_text(encoding="utf-8"))
+        ontology_structure = OntologyStructure.from_dict(ontology_dict)
+        return self.user_client.create_ontology(title, structure=ontology_structure, description=description)
+
+    @staticmethod
+    def prepare_label_row(
+        original_label_row: dict, initiated_label_row: dict, new_label_row: dict, original_du: str
```

Comment:
[nitpick] 💅 

just guessing the types by the names, at least the ones that have a type

```suggestion
        original_label_row: LabelRow, initiated_label_row: LabelRow, new_label_row: LabelRow, original_du: dict
```